File: Programming Fundamentals with Python/4. Functions/Exercise/04_odd_and_even_sum.py
def odd_and_even_sum(str_nums):
    int_nums = [int(num) for num in str_nums]
    sum_of_odd_digits = sum(list(filter(lambda x: x % 2 != 0, int_nums)))
    sum_of_even_digits = sum(list(filter(lambda x: x % 2 == 0, int_nums)))
    print(f"Odd sum = {sum_of_odd_digits}, Even sum = {sum_of_even_digits}")

numbers = input()
odd_and_even_sum(numbers)

File: JS Front-End/2. Functions and Statements/Exercises/04. Odd And Even Sum.js
function oddAndEvenSum(num) {
    let arrNum = String(num)
    .split("")
    .map((ch) => Number(ch));
     
    let sum = function(arrNum) {
        let odds = 0;
        let evens = 0;

        for (const n of arrNum) {
            if (n % 2 == 0) {
                evens += n;
            } else {
                odds += n;
            }
        }
        return [odds, evens]
    }
    let [oddSum, evenSum] = sum(arrNum);
    return `Odd sum = ${oddSum}, Even sum = ${evenSum}`
}


// Test
console.log(oddAndEvenSum(1000435) === "Odd sum = 9, Even sum = 4");
console.log(oddAndEvenSum(3495892137259234) === "Odd sum = 54, Even sum = 22");
console.log(oddAndEvenSum(1000435));
console.log(oddAndEvenSum(3495892137259234));
